extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java

```diff
@@ -207,7 +210,40 @@ public static void addConsumerPropertiesFromConfig(
     ClassLoader currCtxCl = Thread.currentThread().getContextClassLoader();
     try {
       Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
-      return new KafkaConsumer<>(props, new ByteArrayDeserializer(), new ByteArrayDeserializer());
+      Deserializer keyDeserializerObject;
+      Deserializer valueDeserializerObject;
+  
+      try {
+        Class keyDeserializerClass = Class.forName(props.getProperty("key.deserializer", "org.apache.kafka.common.serialization.ByteArrayDeserializer"));
+        Method keyDeserializerMethod = keyDeserializerClass.getMethod("deserialize", String.class, byte[].class);
+        Type keyDeserializerReturnType = keyDeserializerMethod.getGenericReturnType();
+    
+        if (keyDeserializerReturnType.getTypeName().equals("byte[]")) {
+          keyDeserializerObject = (Deserializer) keyDeserializerClass.getConstructor().newInstance();
+        } else {
+          throw new IllegalArgumentException("Key deserializer must return a byte array (byte[]), " + keyDeserializerClass.getName() + " returns " + keyDeserializerReturnType.getTypeName());
+        }
+      }
+      catch (ClassNotFoundException | NoSuchMethodException | InstantiationException | IllegalAccessException | InvocationTargetException e) {
+        throw new StreamException(e);
+      }
+  
+      try {
+        Class valueDeserializerClass = Class.forName(props.getProperty("value.deserializer", "org.apache.kafka.common.serialization.ByteArrayDeserializer"));
+        Method valueDeserializerMethod = valueDeserializerClass.getMethod("deserialize", String.class, byte[].class);
+        Type valueDeserializerReturnType = valueDeserializerMethod.getGenericReturnType();
+    
+        if (valueDeserializerReturnType.getTypeName().equals("byte[]")) {
+          valueDeserializerObject = (Deserializer) valueDeserializerClass.getConstructor().newInstance();
+        } else {
+          throw new IllegalArgumentException("Key deserializer must return a byte array (byte[]), " + valueDeserializerClass.getName() + " returns " + valueDeserializerReturnType.getTypeName());
```

Comment:
Should say "Value deserializer".

Reply:
caught red-handed with a copy and paste error.  Fixed.